staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -866,7 +871,14 @@ func StrategicMergeMapPatch(original, patch JSONMap, dataStruct interface{}) (JS
 	// each field in a strategic merge patch, we can't find the go struct tags. Hence, we can't easily  do a strategic merge
 	// for custom resources. So we should fail fast and return an error.
 	if _, ok := dataStruct.(*unstructured.Unstructured); ok {
-		return nil, mergepatch.ErrUnsupportedStrategicMergePatchFormat
+		if _, ok := patch["metadata"]; !ok || len(patch) != 1 {
```

Comment:
@sttts I forgot to push this check. This will prevent strategic merge patch outside of "metadata".

Reply:
is an empty patch ever allowed?

Reply:
@jennybuckley An empty strategic merge patch on custom resources is not allowed (It wasn't allowed before as well).